### ISSUE TITLE
ERR_TLS_CERT_ALTNAME_INVALID

### ISSUE THREAD

**Issue by Robbilie** [xmppjs/xmpp.js#533]
when trying to connect to my server or even the one in your code used as an example i get this error:
```
Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: localhost. is not in the cert's altnames: DNS:*.olark.com, DNS:olark.com
```

any ideas?
PS: that PR of yours about transports parameters would be greatly appreciated

**Comment by sonnyp** [comment]
Well the error is pretty self explanatory.

It's an issue with your certificate, not with xmpp.js .

What domain/host are you trying to connect to? Do you want to connect using TLS or plain?

**Comment by Robbilie** [comment]
so im currently working on a different project and just had to create a socket using tls.connect. and i received the same error and fixed it by passing not only the socket but also the host to said method. et voila, it works. turns out, you have the same method call and you too only pass it the socket. might want to consider looking into that

**Comment by sonnyp** [comment]
@Robbilie indeed, my bad, fixed here https://github.com/xmppjs/xmpp.js/pull/579

**Comment by sonnyp** [comment]
fixed in 0.5.2
